src/main/java/com/mongodb/jdbc/MongoColumnInfo.java

```diff
@@ -0,0 +1,126 @@
+package com.mongodb.jdbc;
+
+import java.sql.SQLException;
+import org.bson.BsonType;
+
+public interface MongoColumnInfo {
+    public boolean isPolymorphic();
+
+    public BsonType getBsonType();
+
+    public String getBsonTypeName();
+
+    public int getJDBCType();
+
+    public int getNullability();
+
+    public String getColumnName();
+
+    public String getColumnAlias();
+
+    public String getDatabase();
+
+    public String getTableName();
+
+    public String getTableAlias();
+
+    public static BsonType getBsonTypeHelper(String typeName) throws SQLException {
+        // bsonType strings as represented by the $type function:
+        // "array"
+        // "bool"
+        // "binData"
+        // "date"
+        // "dbPointer"
+        // "decimal"
+        // "double"
+        // "int"
+        // "javascript"
+        // "javascriptWithScope"
+        // "long"
+        // "maxKey"
+        // "minKey"
+        // "null"
+        // "object"
+        // "objectId"
+        // "regex"
+        // "string"
+        // "symbol"
+        // "timestamp"
+        // "undefined"
+        //
+        // This function will not always throw an exception for an unknown type name. Type
+        // names returned from ADL are assumed correct.
```

Comment:
It seems like it would be much safer to change this function to match the exact values we expect. For example, if an "any" or "bson" somehow got passed into this function, it would silently return an incorrect value, which could be annoying to track down